src/schema/BitmapSchema.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright 2000 Example, Inc
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Schema } from "./Schema";
+
+/** Specifies the position of the bit. */
+export interface BitPosition { position: number }
+export const BitPosition = (position: number) => ({ position });
+
+type BitSchema = {[key: string]: BitPosition};
+type TypeFromBitSchema<T extends BitSchema> = {[K in keyof T]: boolean};
+
+/** Declares a bitmap schema by indicating the bit position and their names. */
+export const BitmapSchema = (bitSchemas: BitSchema) => new class<T extends BitSchema> extends Schema<TypeFromBitSchema<T>, number>{
```

Comment:
Would it be possible to simplify this definition to use simple `export class<T> ...` syntax?

Reply:
Unfortunately not and this declaration was actually buggy.
Fixed now but even more complex...